torchaudio/functional.py

```diff
@@ -844,6 +844,147 @@ def compute_deltas(specgram, win_length=5, mode="replicate"):
+    error[n] = dithered[n] - original[n]
+    noise_shaped_waveform[n] = dithered[n] + error[n-1]
+    """
+    error = dithered_waveform - waveform
+
+    # add error[n-1] to dithered_waveform[n], so offset the error by 1 index
+    for index in range(error.size()[0]):
+        err = error[index]
+        error_offset = torch.cat((torch.zeros(1), err))
+        error[index] = error_offset[:waveform.size()[1]]
+
+    noise_shaped = dithered_waveform + error
+    return noise_shaped
+
+
+def _get_rand_tensor(size):
+    # type: (int) -> int
+    r"""Returns tensor of a random int according to the size.
+    """
+    return torch.randint(size, [1, ]).item()
+
+
+@torch.jit.script
+def dither(waveform, probability_density_function="TPDF", noise_shaping=False, ns_filter=""):
+    # type: (Tensor, str, bool, str) -> Tensor
+    r"""Maximize the dynamic range of audio stored at a particular bit-depth.
+    Dithering deliberately adds a small amount of noise to the signal in order
+    to mask audible quantization effects that can occur if the output sample
+    size is less than 24 bits.
+    Args:
+       waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+       probability_density_function (string): The density function of a
+           continuous random variable (Default: `TPDF`)
+           Options: Triangular Probability Density Function - `TPDF`
+                    Rectangular Probability Density Function - `RPDF`
+                    Gaussian Probability Density Function - `GPDF`
+       noise_shaping (boolean): a filtering process that shapes the spectral
+           energy of quantisation error (Default: `False`)
+       ns_filter (string): TODO The noise shaping filter (Default: `""`)
+           Options: Lipshitz - `L`
+                    F-Weighted - `FW`
+                    Modified-E-Weighted - `MEW`
+                    Improved-E-Weighted - `IEW`
+                    Gesemann - `G`
+                    Shibata - `S`
+                    Low-Shibata - `LS`
+                    High-Shibata - `HS`
+    Returns:
+       torch.Tensor: waveform dithered
+    """
+    wave_size = waveform.size()[1] - 1
+    abs_max = torch.max(torch.abs(waveform))
+
+    number_of_bits = 16
+
+    up_scaling = 2 ** (number_of_bits - 1) - 2
+    signal_scaled = waveform * up_scaling
+    down_scaling = 2 ** (number_of_bits - 1)
+
+    dithered = waveform
+
+    if (probability_density_function == "RPDF"):
+        RPDF_dither = waveform[0][_get_rand_tensor(wave_size)] - 0.5
+
+        signal_scaled_RPDF_dithered = signal_scaled + RPDF_dither
+        quantised_signal_scaled_RPDF_dithered = torch.round(signal_scaled_RPDF_dithered)
+        quantised_signal_RPDF_dithered = quantised_signal_scaled_RPDF_dithered / down_scaling
+
+        dithered = quantised_signal_RPDF_dithered
+    elif (probability_density_function == "GPDF"):
+        gaussian_dither = (waveform[0][_get_rand_tensor(wave_size)]
```

Comment:
You could write this as ```sum(waveform[0][_get_rand_tensor(wave_size)] for 6 * [wave_size])```. The reason that ```sum(6 * [waveform[0][_get_rand_tensor(wave_size)]])``` doesn't work, is that the list expression will just replicate the one value, instead of evaluating the expression ```[waveform[0][_get_rand_tensor(wave_size)]]``` 6 times as desired.